harness/determined/common/experimental/determined.py

```diff
@@ -155,32 +155,36 @@ def create_model(
 
         return model.Model.from_json(r.json().get("model"), self._session)
 
-    def get_model(self, model_id: int) -> model.Model:
+    def get_model(self, name: str) -> model.Model:
         """
         Get the :class:`~determined.experimental.Model` from the model registry
-        with the provided id. If no model with that id is found in the registry,
+        with the provided name. If no model with that id is found in the registry,
         an exception is raised.
         """
-        r = self._session.get("/api/v1/models/{}".format(model_id))
+        r = self._session.get("/api/v1/models/{}".format(name))
         return model.Model.from_json(r.json().get("model"), self._session)
 
-    def get_model_by_name(self, name: str) -> model.Model:
+    def get_model_by_id(self, model_id: int) -> model.Model:
         """
         Get the :class:`~determined.experimental.Model` from the model registry
-        with the provided name. If no model with that name is found in the registry,
+        with the provided id. If no model with that id is found in the registry,
         an exception is raised.
         """
-        r = self._session.get("/api/v1/models?name={}".format(name))
+        r = self._session.get("/api/v1/models?id={}".format(model_id))
         models = r.json().get("models")
         assert len(models) > 0
         return model.Model.from_json(models[0], self._session)
 
+    def get_model_by_name(self, name: str) -> model.Model:
+        return self.get_model(name)
+
     def get_models(
         self,
         sort_by: model.ModelSortBy = model.ModelSortBy.NAME,
         order_by: model.ModelOrderBy = model.ModelOrderBy.ASCENDING,
         name: str = "",
         description: str = "",
+        model_id: int = 0,
```

Comment:
do you have a guarantee that `model_id`s are always non-zero?

Reply:
Also... there should never be multiple models with the same model_id, so why isn't `get_model_by_id()` sufficient?

(also, I have the same question for `name`)

Reply:
My experience is that model_id in Postgres starts at 1. A 0 would cause problems elsewhere in our code (proto returns a 0 for an empty int).

I wrote this for `get_model_by_id` to piggy-back onto `get_models`, if they are separated then we need a new API route to accommodate it.

The name param will be included here for backwards compatibility / case-insensitive queries.